### ISSUE TITLE
Tasks fail - unable to locate python modules

### ISSUE THREAD

**Issue by ggg121** [radical-collaboration/slrise#14]
I run, within a virtual environment, "python FACTS.py ./experiments/temp_exp/" (from the restructure_facts branch).  Everything seems to go ok until the individual tasks are executed.  Each task is returned with a "fail" status.  The kopp14-icesheets pre-process stage (one task) produces error output indicating that numpy cannot be found.  Running the associated code outside of FACTS works as expected.  It seems like there's a mismatch in Python environments between running code inside versus outside EnTK.

**Comment by mturilli** [comment]
@ggg121 , this seems to be an issue with the runtime (RP). Could you do the following and then share the result:
```
RADICAL_VERBOSE=DEBUG RADICAL_ENTK_VERBOSE=DEBUG <python your_script_here.py> &> verbose.log
radicalpilot-fetch-logfiles <session id>
```

In order to determine the session id, you can look for a folder that is created on the client in current working directory. It should have the format ```rp.session.*```. You can find the latest folder by doing ``ls -ltr`` (last is recent). All the logfiles are brought to this rp.session.* folder. Please zip this folder and attach to github ticket.

**Comment by ggg121** [comment]
Attached are the `verbose.log` file and a zipped archive of `re.session.*`.  I've also attached the error file produced by the first task that fails `kopp_icesheets-preprocess_task1_err.txt`.  Subsequent task error files are not generated.  This last file indicates that python cannot find the Numpy module.  This is why I think there's a mismatch between the Python environments.

Note the command to fetch the log files in the version of EnTK I'm using (0.7.17) is:

`radical-pilot-fecth-logfiles <session id>`

...and once I run that command, the directory is created in its respective `re.session.*` directory.

Also, there are no `rp.session.*` directories, only `re.session.*`.

[verbose.log](https://github.com/radical-collaboration/slrise/files/3302245/verbose.log)
[re.session.ggg46-vb.ggg46.018065.0000.zip](https://github.com/radical-collaboration/slrise/files/3302258/re.session.ggg46-vb.ggg46.018065.0000.zip)
[kopp14-icesheets_pre-process_task1_err.txt](https://github.com/radical-collaboration/slrise/files/3302260/kopp14-icesheets_pre-process_task1_err.txt)

**Comment by mturilli** [comment]
Thanks. There is indeed a problem with numpy but I also see some errors in RP logs. Can I have the command you run so to try to reproduce this myself?

**Comment by ggg121** [comment]
From the root repo directory...

`python FACTS.py ./experiments/temp_exp`

Note:  For the logs I provided in my last comment, I had commented out the full "thermalexpansion" entry in the `./experiments/temp_exp/config.yml` file for troubleshooting.  The same failures occur even if "thermalexpansion" is included.

**Comment by mturilli** [comment]
Thanks @ggg121 , I successfully reproduced your problem. I installed numpy and I get the following error:
```
Traceback (most recent call last):
  File "kopp14_preprocess_icesheets.py", line 81, in <module>
    kopp14_preprocess_icesheets(args.scenario)
  File "kopp14_preprocess_icesheets.py", line 58, in kopp14_preprocess_icesheets
    ratefile = open(os.path.join(outdir, "kopp14_icesheets_rates.pkl"), 'wb')
IOError: [Errno 2] No such file or directory: 'data/kopp14_icesheets_rates.pkl'
```

**Comment by ggg121** [comment]
The rapid fix installs numpy in the radical sandbox environment, which appears to be persistent across multiple EnTK sessions.  I would imagine leaving `pip install numpy` there shouldn't cause problems, but it will be at least checking for a numpy installation each time the kopp14-icesheets module is requested.  I suggest we leave it there for now.

As you pointed out in an earlier discussion, we would likely need to install additional modules.  I'll compile and post a list of python modules currently used in the FACTS modules.

Note this fix also addresses issue #15.  Let's hold off on a merge until we figure out our approach to #15.

**Comment by andre-merzky** [comment]
> The rapid fix installs numpy in the radical sandbox environment, which appears to be persistent across multiple EnTK sessions.

yes.  But note that this approach carries the risk of interfering with EnTK and the underlying layers - on some  machines we depend on rather specific setup options to get our stack to work reliably.  In general it might be better to create a dedicated persistent virtualenv for the workload.

Having said that: if it works, it works :-D

**Comment by ggg121** [comment]
@mturilli - Your rapid fix works for numpy, unfortunately it does not work for scipy.  It looks like the scipy developers are no longer supporting Python 2.x.  Both numpy and scipy are crucial for developing these SLR modules.  Per our discussion earlier, this would all be solved once EnTK is ported to Python 3, but for now we may need to take the approach @andre-merzky mentioned.

Below is a list of the Python modules currently needed to run the SLR modules:
os, sys, argparse, pickle, yaml, numpy, scipy, netCDF4, time, pprint, glob, re, radical.entk

There may be more as development of the SLR modules continues, but these should let us run the kopp14-icesheets and kopp14-thermalexpansion modules (in their current state) start to finish.

**Comment by mturilli** [comment]
Hi @ggg121 , thank you for the list. Is your code in Python 2 or Python 3? We can force the installation of scipy for Python 2 if the former.

**Comment by ggg121** [comment]
Thanks @mturilli for looking into this.  The code is in Python 2.  I don't think it would take much effort to port my code to Python 3...the only hurdle at the moment would be EnTK.

**Comment by mturilli** [comment]
I pushed a fix for this issue to branch `fix/issue14-15`. I successfully executed tasks 0, 1, 2 getting `kopp14_icesheets_fit.pkl` and `kopp14_icesheets_projections.pkl`. I think there was an issue with post processing of Task1 but it did not seem related to this ticket or the RADICAL stack.

**Comment by ggg121** [comment]
Thanks @mturilli for the fix.  I ran into the same problem you describe, just a different reason.  You're likely missing the `pyssht` module, which is something (as far as I know) that isn't configured to be installed via pip.  I have it installed on my laptop, so the problem I had was `matplotlib` was missing.  I added it to the pre-exec list of modules and all four stages of the K14 icesheets module ran without a problem.

I'll merge the `fix/issue14-15` branch into master with these changes.
